lib.php

```diff
@@ -844,3 +884,47 @@ function mod_zoom_get_fontawesome_icon_map() {
         'mod_zoom:i/calendar' => 'fa-calendar'
     ];
 }
+
+/**
+ * This function updates the tracking field settings in config_plugins.
+ */
+function mod_zoom_update_tracking_fields() {
+    global $DB;
+
+    $defaulttrackingfields = zoom_clean_tracking_fields();
+    $zoomtrackingfields = zoom_list_tracking_fields();
+    $zoomprops = array('id', 'field', 'required', 'visible', 'recommended_values');
+    $confignames = array();
+
+    foreach ($defaulttrackingfields as $key => $defaulttrackingfield) {
+        if ($defaulttrackingfield !== '') {
+            foreach ($zoomtrackingfields as $zoomtrackingfield) {
+                if (in_array($defaulttrackingfield, $zoomtrackingfield, true)) {
+                    foreach ($zoomprops as $zoomprop) {
+                        $configname = 'tf_' . $key . '_' . $zoomprop;
+                        $confignames[] = $configname;
+                        if ($zoomprop === 'recommended_values') {
+                            $configvalue = implode(', ', $zoomtrackingfield[$zoomprop]);
+                        } else {
+                            $configvalue = $zoomtrackingfield[$zoomprop];
+                        }
+                        set_config($configname, $configvalue, 'zoom');
+                    }
+                    break;
+                }
+            }
+        }
+    }
```

Comment:
Not sure if I understood this 100%, but if I did and `field` is the only intended match, then this should do the right thing:
```suggestion
    foreach ($zoomtrackingfields as $zoomtrackingfield) {
        $key = strtolower($zoomtrackingfield->field);
        if (isset($defaulttrackingfields[$key])) {
            foreach ($zoomprops as $zoomprop) {
                $configname = 'tf_' . $key . '_' . $zoomprop;
                $confignames[] = $configname;
                if ($zoomprop === 'recommended_values') {
                    $configvalue = implode(', ', $zoomtrackingfield[$zoomprop]);
                } else {
                    $configvalue = $zoomtrackingfield[$zoomprop];
                }
                set_config($configname, $configvalue, 'zoom');
            }
        }
    }
```

Reply:
Going beyond that, assuming I did understand this correctly, as long as the `field` values are "unique", then we could consider moving the `$key = strtolower($zoomtrackingfield->field);` line into the zoom_list_tracking_fields() function.